gcloud-java-datastore/src/main/java/com/google/gcloud/datastore/testing/LocalDatastoreHelper.java

```diff
@@ -589,80 +540,102 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
     });
   }
 
-  public static LocalGcdHelper start(String projectId, int port, double consistency)
-      throws IOException, InterruptedException {
-    LocalGcdHelper helper = new LocalGcdHelper(projectId, port);
-    helper.start(consistency);
-    return helper;
+  private LocalDatastoreHelper(double consistency) {
+    projectId = PROJECT_ID_PREFIX + UUID.randomUUID().toString();
+    this.consistency = consistency;
+    this.port = findAvailablePort();
   }
 
-  public static void main(String... args) throws IOException, InterruptedException {
-    Map<String, String> parsedArgs = parseArgs(args);
-    String action = parsedArgs.get("action");
-    int port =
-        (parsedArgs.get("port") == null) ? DEFAULT_PORT : Integer.parseInt(parsedArgs.get("port"));
-    switch (action) {
-      case "START":
-        if (!isActive(DEFAULT_PROJECT_ID, port)) {
-          double consistency = parsedArgs.get("consistency") == null
-              ? DEFAULT_CONSISTENCY : Double.parseDouble(parsedArgs.get("consistency"));
-          LocalGcdHelper helper = start(DEFAULT_PROJECT_ID, port, consistency);
-          try (FileWriter writer = new FileWriter(".local_gcd_helper")) {
-            writer.write(helper.gcdPath.toAbsolutePath().toString() + System.lineSeparator());
-            writer.write(Integer.toString(port));
-          }
-        }
-        return;
-      case "STOP":
-        File file = new File(".local_gcd_helper");
-        String path = null;
-        boolean fileExists = file.exists();
-        if (fileExists) {
-          try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
-            path = reader.readLine();
-            port = Integer.parseInt(reader.readLine());
-          }
-        }
-        sendQuitRequest(port);
-        if (fileExists) {
-          deleteRecurse(Paths.get(path));
-          file.delete();
-        }
-        return;
-      default:
-        break;
+  private static int findAvailablePort() {
```

Comment:
Let's create an issue for creating a base class for test helpers.
This is an example of code that could be shared.